internal/uvm/vpmem_mapped.go

```diff
@@ -64,33 +64,36 @@ func pageAlign(t uint64) uint64 {
 	return (t/PageSize + 1) * PageSize
 }
 
-// fileSystemSize retrieves ext4 fs SuperBlock and calculates the size of the actual file system
-func fileSystemSize(vhdPath string) (uint64, error) {
-	sb, err := tar2ext4.ReadExt4SuperBlock(vhdPath)
-	if err != nil {
-		return 0, err
-	}
-	blockSize := uint64(1024 * (1 << sb.LogBlockSize))
-	fsSize := blockSize * uint64(sb.BlocksCountLow)
-	return pageAlign(fsSize), nil
-}
-
 // newMappedVPMemModifyRequest creates an hcsschema.ModifySettingsRequest to modify VPMem devices/mappings
 // for the multi-mapping setup
 func newMappedVPMemModifyRequest(ctx context.Context, rType string, deviceNumber uint32, md *mappedDeviceInfo, uvm *UtilityVM) (*hcsschema.ModifySettingRequest, error) {
+	guestSettings := guestrequest.LCOWMappedVPMemDevice{
+		DeviceNumber: deviceNumber,
+		MountPath:    md.uvmPath,
+		MappingInfo: &guestrequest.LCOWMappedLayer{
+			DeviceOffsetInBytes: md.mappedRegion.Offset(),
+			DeviceSizeInBytes:   md.sizeInBytes,
+		},
+	}
+
+	if verity, err := readVeritySuperBlock(ctx, md.hostPath); err != nil {
+		log.G(ctx).WithError(err).WithField("hostPath", md.hostPath).Debug("unable to read dm-verity information from VHD")
+	} else {
+		if verity != nil {
```

Comment:
Is this verity != nil check needed given err != nil. It didn't appear to be.

Reply:
you're right, this is not needed.